src/main/java/org/jabref/gui/preftabs/AdvancedTab.java

```diff
@@ -156,10 +158,12 @@ public boolean validateSettings() {
                 throw new NumberFormatException();
             }
         } catch (NumberFormatException ex) {
-            JOptionPane.showMessageDialog(null,
+
+            frame.getDialogService().showErrorDialogAndWait(Localization.lang("Remote server port"),
                     Localization.lang("You must enter an integer value in the interval 1025-65535 in the text field for")
                             + " '" + Localization.lang("Remote server port") + '\'',
-                    Localization.lang("Remote server port"), JOptionPane.ERROR_MESSAGE);
+                    ex);
```

Comment:
the exception as an argument is unneeded here, since the stack trace is not helpful at all in this situation: the user entered data in the wrong format and this is said already in the error message.